include/MappingResults.hpp

```diff
@@ -88,7 +102,15 @@ struct MappingResults {
         stats["WCNF"] = wcnf;
       }
     } else if (config.method == Method::Heuristic) {
-      stats["teleportations"] = output.teleportations;
+      stats["teleportations"]      = output.teleportations;
+      auto& benchmark              = stats["benchmark"];
+      benchmark["expanded_nodes"]  = heuristicBenchmark.expandedNodes;
+      benchmark["generated_nodes"] = heuristicBenchmark.generatedNodes;
+      benchmark["time_per_node"]   = heuristicBenchmark.timePerNode;
+      benchmark["average_branching_factor"] =
+          heuristicBenchmark.averageBranchingFactor;
+      benchmark["effective_branching_factor"] =
+          heuristicBenchmark.effectiveBranchingFactor;
```

Comment:
As I also mentioned in the PR description above, `solutionDepth` is only being tracked for each layer individually. I believe a global solution depth does not make too much sense. 
Implementing it as the sum of all layer-wise solution depths would be misleading, as running multiple separate search instances is not the same as running one with the sum of their solution depths (would result in much larger trees and therefore different effective branching rates). 
Implementing it as an average of all layers would make more sense, but would require another data type (floating point instead of integer) and would also be slightly confusing as the only global value with an average, that is not weighted by search tree size (which itself also would not make too much sense for the solution depth).
So I just decided to omit the solution depth for the global case.

Reply:
Thanks for the explanation! That makes total sense. Missed that when I looked over the PR.